app/assets/javascripts/components/course_creator/course_creator.jsx

```diff
@@ -65,7 +65,12 @@ const CourseCreator = React.createClass({
 
   storeDidChange() {
     this.setState(getState());
-    this.state.tempCourseId = CourseUtils.generateTempId(this.state.course);
+    if (this.state.course.term !== '' && this.state.course.school !== '' && this.state.title !== '') {
```

Comment:
So this condition should execute only when this.state.default_course_type === 'ClassroomProgramCourse'  and the trailing _() removed in the else section....
OR do I need to make changes in the course_utils.js